format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatConfig.java

```diff
@@ -0,0 +1,211 @@
+    "Can be either sheet name or sheet no; for example: 'Sheet1' or '0' in case user selects 'Sheet Name' or " +
+    "'Sheet Number' as 'sheet' input respectively. Sheet number starts with 0.";
+  public static final String DESC_TERMINATE_ROW = "Specify whether processing needs to be terminated in case an" +
+    " empty row is encountered while processing excel files. Default value is false.";
+  public static final Map<String, PluginPropertyField> XLS_FIELDS;
+
+  static {
+    Map<String, PluginPropertyField> fields = new HashMap<>(FIELDS);
+    fields.put(NAME_SKIP_HEADER,
+            new PluginPropertyField(NAME_SKIP_HEADER, DESC_SKIP_HEADER, "boolean", false, true));
+    // Add fields specific for excel format handling.
+    fields.put(NAME_SHEET, new PluginPropertyField(NAME_SHEET, DESC_SHEET, "string", false, true));
+    fields.put(NAME_SHEET_VALUE, new PluginPropertyField(NAME_SHEET_VALUE, DESC_SHEET_VALUE, "string", false, true));
+    fields.put(NAME_TERMINATE_IF_EMPTY_ROW, new PluginPropertyField(
+                       NAME_TERMINATE_IF_EMPTY_ROW, DESC_TERMINATE_ROW, "boolean", false, true));
+    XLS_FIELDS = Collections.unmodifiableMap(fields);
+  }
+
+  @Macro
+  @Nullable
+  @Name(NAME_SHEET)
+  @Description(DESC_SHEET)
+  private String sheet;
+
+  @Macro
+  @Nullable
+  @Name(NAME_SHEET_VALUE)
+  @Description(DESC_SHEET_VALUE)
+  private String sheetValue;
+
+
+  @Macro
+  @Nullable
+  @Name(NAME_SKIP_HEADER)
+  @Description(DESC_SKIP_HEADER)
+  private Boolean skipHeader;
+
+  @Macro
+  @Nullable
+  @Name(NAME_TERMINATE_IF_EMPTY_ROW)
+  @Description(DESC_TERMINATE_ROW)
+  private Boolean terminateIfEmptyRow;
+
+  public XlsInputFormatConfig() {
+    super();
+  }
+
+  @VisibleForTesting
+  public XlsInputFormatConfig(@Nullable String schema, @Nullable String sheet, @Nullable String sheetValue,
+                              @Nullable Boolean skipHeader, @Nullable Boolean terminateIfEmptyRow) {
+    super();
+    this.schema = schema;
+    this.sheet = sheet;
+    this.sheetValue =  sheetValue;
+    this.skipHeader = skipHeader;
+    this.terminateIfEmptyRow = terminateIfEmptyRow;
+  }
+
+  public long getSampleSize() {
+    return Long.parseLong(getProperties().getProperties().getOrDefault(NAME_SAMPLE_SIZE, "1000"));
+  }
+
+  public String getSheet() {
+    return sheet == null ? SHEET_NUMBER : sheet;
+  }
+
+  @Nullable
+  public String getSheetValue() {
+    return sheetValue;
+  }
+
+  public boolean getSkipHeader() {
+    return skipHeader != null ? skipHeader : false;
+  }
+
+  public boolean getTerminateIfEmptyRow() {
+    return terminateIfEmptyRow != null ? terminateIfEmptyRow : false;
+  }
+
+  /**
+   * Parses a list of key-value items of column names and their corresponding data types, manually set by the user.
+   *
+   * @return A hashmap of column names and their manually set schemas.
+   */
+  public Map<String, Schema> getOverride() throws IllegalArgumentException {
```

Comment:
I'm having trouble understanding how this is being used. Do you know what is setting this override property?

In any case, it looks like this is copied directly from DelimitedConfig. It should be defined in a single place and not duplicated.

Reply:
I had a discussing with my senior, this feature is not required in XLS source.
This has been removed !